### PR TITLE
feat(auth): add client auth

### PR DESCRIPTION
**Pull request by lionelB**
WIP client auth and graphql request
- fetchOptionExchange is probably buggy ;)
- need to handle expired token properly (ie refetch token before dogin the request)
- figure how to handle error 
- check how cache perform

### CODE REVIEW

File: src/lib/setRefreshTokenCookie.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import cookie from "cookie";
+
+export function setRefreshTokenCookie(res, refresh_token) {
+  res.setHeader(
+    "Set-Cookie",
+    cookie.serialize("refresh_token", refresh_token, {
+      // sameSite: "lax",
```
Comment (revolunet) [inline src/lib/setRefreshTokenCookie.js:7]: should be strict

File: jsconfig.json [hunk 2]
```diff
@@ -0,0 +1,5 @@
+{
+  "compilerOptions": {
+    "baseUrl": "src"
```
Comment (douglasduteil) [inline jsconfig.json:3]: You might unlock more power with "." :smirk: 

```suggestion
    "baseUrl": "."
```

File: src/components/CustomUrqlClient.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import { dedupExchange, cacheExchange, fetchExchange } from "@urql/core";
+import { withUrqlClient } from "next-urql";
+import { refreshToken } from "../lib/auth";
+import { authExchange, tapExchange } from "lib/authTokenExchange";
```
Comment (douglasduteil) [inline src/components/CustomUrqlClient.js:4]: is "lib" "../lib" ?

File: src/pages/admin.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
+import React from "react";
+import { withCustomUrqlClient } from "../components/CustomUrqlClient";
+import { Role } from "components/Role";
+import { Users } from "components/Users";
```
Comment (douglasduteil) [inline src/pages/admin.js:4]: is `../components` `components` ?

```suggestion
import { withCustomUrqlClient } from "components/CustomUrqlClient";
import { Role } from "components/Role";
import { Users } from "components/Users";
```

File: src/pages/api/login.js [hunk 5]
```diff
@@ -1,12 +1,12 @@
 import Boom from "@hapi/boom";
 import Joi from "@hapi/joi";
 import { verify } from "argon2";
-import cookie from "cookie";
 import { createErrorFor } from "../../../src/lib/apiError";
 import { getExpiryDate } from "../../../src/lib/duration";
 import { graphqlClient } from "../../../src/lib/graphqlClient";
 import { generateJwtToken } from "../../../src/lib/jwt";
 import { loginQuery, refreshTokenMutation } from "./login.gql";
+import { setRefreshTokenCookie } from "lib/setRefreshTokenCookie";
```
Comment (douglasduteil) [inline src/pages/api/login.js:9]:
```suggestion
import { createErrorFor } from "src/lib/apiError";
import { getExpiryDate } from "src/lib/duration";
import { graphqlClient } from "src/lib/graphqlClient";
import { generateJwtToken } from "src/lib/jwt";
import { setRefreshTokenCookie } from "src/lib/setRefreshTokenCookie";
import { loginQuery, refreshTokenMutation } from "./login.gql";
```

**Comment by revolunet** [comment]
je propose qu'on merge déjà ça

**Comment by revolunet** [comment]
@lionelB on peut merger ?

**Pull request merged**